In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt 
import seaborn as sns

Note: Data is from the UCI Machine Learning Repository:

Dua, D. and Graff, C. (2019). UCI Machine Learning Repository [http://archive.ics.uci.edu/ml]. Irvine, CA: University of California, School of Information and Computer Science.

In [3]:
# data: https://archive.ics.uci.edu/ml/datasets/heart+disease
heart = pd.read_csv('processed.cleveland.data.csv')

In [4]:
heart.head()

age  sex   cp  trestbps   chol  fbs  restecg  thalach  exang  oldpeak  \
0  63.0  1.0  1.0     145.0  233.0  1.0      2.0    150.0    0.0      2.3   
1  67.0  1.0  4.0     160.0  286.0  0.0      2.0    108.0    1.0      1.5   
2  67.0  1.0  4.0     120.0  229.0  0.0      2.0    129.0    1.0      2.6   
3  37.0  1.0  3.0     130.0  250.0  0.0      0.0    187.0    0.0      3.5   
4  41.0  0.0  2.0     130.0  204.0  0.0      2.0    172.0    0.0      1.4   

   slope   ca thal  heart_disease  
0    3.0  0.0  6.0              0  
1    2.0  3.0  3.0              2  
2    2.0  2.0  7.0              1  
3    3.0  0.0  3.0              0  
4    1.0  0.0  3.0              0

- age: age in years
- sex: 1=male, 0=female
- cp: chest pain type
 - Value 1: typical angina
 - Value 2: atypical angina
 - Value 3: non-anginal pain
 - Value 4: asymptomatic
- trestbps: resting blood pressure (in mm Hg on admission to the hospital)
- chol: serum cholestoral in mg/dl
- fbs: (fasting blood sugar > 120 mg/dl) (1 = true; 0 = false)
- restecg: resting electrocardiographic results
 - Value 0: normal
 - Value 1: having ST-T wave abnormality (T wave inversions and/or ST elevation or depression of > 0.05 mV) 
 - Value 2: showing probable or definite left ventricular hypertrophy by Estes' criteria
- thalach: maximum heart rate achieved in an exercise test
- exang: exercise induced angina (1 = yes; 0 = no)
- oldpeak: ST depression induced by exercise relative to rest
- slope: the slope of the peak exercise ST segment
 - Value 1: upsloping
 - Value 2: flat
 - Value 3: downsloping
- ca: number of major vessels (0-3) colored by flourosopy
- thal: 
 - Value 3: normal
 - Value 6: fixed defect
 - Value 7: reversable defect
- heart_disease: diagnosis of heart disease (angiographic disease status)
 - Value 0: < 50% diameter narrowing
 - Value 1: > 50% diameter narrowing
"\[This field\] refers to the presence of heart disease in the patient. It is integer valued from 0 (no presence) to 4. Experiments with the Cleveland database have concentrated on simply attempting to distinguish presence (values 1,2,3,4) from absence (value 0)."


In [7]:
heart.info()

heart.describe(include='all')


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 303 entries, 0 to 302
Data columns (total 14 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   age            303 non-null    float64
 1   sex            303 non-null    float64
 2   cp             303 non-null    float64
 3   trestbps       303 non-null    float64
 4   chol           303 non-null    float64
 5   fbs            303 non-null    float64
 6   restecg        303 non-null    float64
 7   thalach        303 non-null    float64
 8   exang          303 non-null    float64
 9   oldpeak        303 non-null    float64
 10  slope          303 non-null    float64
 11  ca             303 non-null    object 
 12  thal           303 non-null    object 
 13  heart_disease  303 non-null    int64  
dtypes: float64(11), int64(1), object(2)
memory usage: 33.3+ KB


age         sex          cp    trestbps        chol  \
count   303.000000  303.000000  303.000000  303.000000  303.000000   
unique         NaN         NaN         NaN         NaN         NaN   
top            NaN         NaN         NaN         NaN         NaN   
freq           NaN         NaN         NaN         NaN         NaN   
mean     54.438944    0.679868    3.158416  131.689769  246.693069   
std       9.038662    0.467299    0.960126   17.599748   51.776918   
min      29.000000    0.000000    1.000000   94.000000  126.000000   
25%      48.000000    0.000000    3.000000  120.000000  211.000000   
50%      56.000000    1.000000    3.000000  130.000000  241.000000   
75%      61.000000    1.000000    4.000000  140.000000  275.000000   
max      77.000000    1.000000    4.000000  200.000000  564.000000   

               fbs     restecg     thalach       exang     oldpeak  \
count   303.000000  303.000000  303.000000  303.000000  303.000000   
unique         NaN         NaN         NaN         NaN         NaN   
top            NaN         NaN         NaN         NaN         NaN   
freq           NaN         NaN         NaN         NaN         NaN   
mean      0.148515    0.990099  149.607261    0.326733    1.039604   
std       0.356198    0.994971   22.875003    0.469794    1.161075   
min       0.000000    0.000000   71.000000    0.000000    0.000000   
25%       0.000000    0.000000  133.500000    0.000000    0.000000   
50%       0.000000    1.000000  153.000000    0.000000    0.800000   
75%       0.000000    2.000000  166.000000    1.000000    1.600000   
max       1.000000    2.000000  202.000000    1.000000    6.200000   

             slope   ca thal  heart_disease  
count   303.000000  303  303     303.000000  
unique         NaN    5    4            NaN  
top            NaN  0.0  3.0            NaN  
freq           NaN  176  166            NaN  
mean      1.600660  NaN  NaN       0.937294  
std       0.616226  NaN  NaN       1.228536  
min       1.000000  NaN  NaN       0.000000  
25%       1.000000  NaN  NaN       0.000000  
50%       2.000000  NaN  NaN       0.000000  
75%       2.000000  NaN  NaN       2.000000  
max       3.000000  NaN  NaN       4.000000